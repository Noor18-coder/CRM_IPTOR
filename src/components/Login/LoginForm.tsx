/* eslint-disable consistent-return */
import React from 'react';
import { Dispatch } from 'redux';
import { Form, Image } from 'react-bootstrap';
import { useSelector, useDispatch } from 'react-redux';

import { Redirect } from 'react-router';
import logo from '../../assets/images/iptor-logo-orange.svg';
import errorIcon from '../../assets/images/error.png';

import LoginFooter from './Shared/LoginFooter';
import { SubmitButton } from './Shared/SubmitButton';
import { CompanySelection } from './CompanySelection';
import LeftColmData from './Shared/LeftColmData';

import { authentication, authWithCompany, logOutSuccess } from '../../store/Auth/Actions';
import { AuthRequest } from '../../store/Auth/Types';
import { AppState } from '../../store/store';

import Loader from '../Shared/Loader/Loader';
import i18n from '../../i18n';

const LoginForm: React.FC = () => {
  const state: AppState = useSelector((loggedState: AppState) => loggedState);
  const [authRequest, setValues] = React.useState<AuthRequest>();
  const [company, setCompany] = React.useState<string>('');
  const [error, setError] = React.useState<boolean>(false);
  const [userError, setUserError] = React.useState<boolean>(false);
  const [passwordError, setPasswordError] = React.useState<boolean>(false);
  const [isSubmit, setIsSubmit] = React.useState<boolean>(false);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (e.currentTarget.value === '') {
      setIsSubmit(false);
    }
    setValues({
      ...authRequest,
      [e.currentTarget.id]: e.currentTarget.value,
    });
    setError(false);
    if (e.currentTarget.value === '') {
      setIsSubmit(false);
    } else if (authRequest && authRequest.user && authRequest.password) {
      setIsSubmit(true);
    }
  };

  const dispatch: Dispatch<any> = useDispatch();
  const doClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    dispatch(authentication(authRequest));
  };

  const selectCompany = (newCompany: string) => {
    setCompany(newCompany);
  };

  const backToLogin = () => {
    setIsSubmit(false);
    dispatch(logOutSuccess());
  };

  const validateField = (_event: any) => {
    const user = document.getElementById('user') as HTMLInputElement;
    const password = document.getElementById('password') as HTMLInputElement;
    if (_event.key.length === 9 && _event.key.length !== 3) {
      if (user.value === '') {
        user.style.border = '1px solid #ED2024';
        setUserError(true);
      }

      if (password.value === '') {
        password.style.border = '1px solid #ED2024';
        setPasswordError(true);
      }
      setIsSubmit(false);
      setError(true);
    }
  };
  React.useEffect(() => {
    if (company) {
      const params = { ...authRequest, company };
      dispatch(authWithCompany(params));
    }
    const user = document.getElementById('user') as HTMLInputElement;
    const password = document.getElementById('password') as HTMLInputElement;

    setTimeout(() => {
      if (user && password) {
        if (user.matches(':-internal-autofill-selected') && password.matches(':-internal-autofill-selected')) {
          setIsSubmit(true);
        }
      }
    }, 100);
  }, [company]);

  const user = document.getElementById('user') as HTMLInputElement;
  const password = document.getElementById('password') as HTMLInputElement;

  setTimeout(() => {
    if (user && password) {
      if (user.matches(':-internal-autofill-selected') && password.matches(':-internal-autofill-selected')) {
        setIsSubmit(true);
      }
    }
  }, 10);

  return (
    <>
      {(() => {
        if (state.auth.login) {
          return <Redirect to="/" />;
        }
        if (state.auth.loginWithoutCompany) {
          return <CompanySelection selectCompany={selectCompany} backToLogin={backToLogin} />;
        }
        return (
          <div className="main-wrapper loginpage">
            <LeftColmData />
            <>
              <div className="login-panel-container">
                <Image src={logo} className="login-form-logo" alt="Iptor" title="Iptor" />
                <p className="welcome-txt">
                  <span className="hi-txt">{i18n.t('hiMessage')}</span>
                  <span className="welcomeback-txt">{i18n.t('welcomeMessage')}</span>
                </p>
                {state.enviornmentConfigs.loadingMask && <Loader component="opportunity" />}
                {error ? (
                  <p className="error">
                    {' '}
                    <Image className="alert-icon" src={errorIcon} width={15} height={12} />
                    &nbsp; {i18n.t('loginErrorMsg')}
                  </p>
                ) : null}
                {state.auth.error ? (
                  <p className="error">
                    {' '}
                    <Image className="alert-icon" src={errorIcon} width={15} height={12} />
                    &nbsp; {state.auth.error}
                  </p>
                ) : null}
                <div className="form-placeholder-container">
                  <Form>
                    <Form.Group>
                      {userError || state.auth.error ? <Form.Label className="err-text">User ID</Form.Label> : <Form.Label>User ID</Form.Label>}
                      <Form.Control
                        className="form-control"
                        placeholder="Enter UserId"
                        id="user"
                        type="text"
                        onChange={handleChange}
                        onKeyDown={validateField}
                      />
                    </Form.Group>

                    <Form.Group>
                      {passwordError || state.auth.error ? <Form.Label className="err-text">Password</Form.Label> : <Form.Label>Password</Form.Label>}
                      <Form.Control
                        className="form-control"
                        placeholder="Enter password"
                        type="password"
                        id="password"
                        onChange={handleChange}
                        onKeyDown={validateField}
                      />
                    </Form.Group>
                    {/* <Nav.Item className="forgotpswd">
                        <Nav.Link href="#">Forgot Password?</Nav.Link>
                      </Nav.Item> */}
                    <SubmitButton title="Login" isSubmit={isSubmit} onClick={doClick} />
                  </Form>
                </div>
                <LoginFooter />
              </div>
            </>
          </div>
        );
      })()}
    </>
  );
};

export default LoginForm;
