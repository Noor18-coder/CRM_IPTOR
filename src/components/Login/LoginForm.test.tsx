import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';
import StoreMock from '../../mocks/Store.mock';

const store = StoreMock.createInitialStore();

describe('[Login] LoginForm', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <LoginForm />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
