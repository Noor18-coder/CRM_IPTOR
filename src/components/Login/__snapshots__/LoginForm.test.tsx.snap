// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Login] LoginForm should renders correctly 1`] = `
<div
  className="main-wrapper loginpage"
>
  <div
    className="login-lft-cont-container"
  >
    <img
      alt="Iptor"
      className="login-cont-logo"
      src="iptor-logo-white.svg"
      title="Iptor"
    />
    <h1
      className="para"
    >
      A milestone in the Iptor transformation journey
    </h1>
    <p
      className="version-details"
    >
      Version 11 is the key platform for distribution and publishing businesses to turn digital disruption into digital opportunity.
    </p>
  </div>
  <div
    className="login-panel-container"
  >
    <img
      alt="Iptor"
      className="login-form-logo"
      src="iptor-logo-orange.svg"
      title="Iptor"
    />
    <p
      className="welcome-txt"
    >
      <span
        className="hi-txt"
      >
        Hi there!
      </span>
      <span
        className="welcomeback-txt"
      >
        Welcome back...
      </span>
    </p>
    <div
      className="form-placeholder-container"
    >
      <form
        className=""
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            User ID
          </label>
          <input
            className="form-control form-control"
            id="user"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Enter UserId"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Password
          </label>
          <input
            className="form-control form-control"
            id="password"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Enter password"
            type="password"
          />
        </div>
        <button
          className="iptor-login-btn disable-button"
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
    <p
      className="login-copyrights"
    >
      Copyright Â© 2021 - Iptor Sweden AB |
       
      <button
        className="text-link footer-anchor-button"
        type="button"
      >
        Privacy Policy
      </button>
       
      |
       
      <button
        className="text-link footer-anchor-button"
        type="button"
      >
        UK Legal?
      </button>
       
      |
       
      <button
        className="text-link footer-anchor-button"
        type="button"
      >
        Code
      </button>
       
    </p>
  </div>
</div>
`;
